processor/intracesampler/fnvhasher.go

```diff
@@ -0,0 +1,37 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package intracesampler // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/intracesamplerprocessor"
+
+import (
+	"hash/fnv"
+)
+
+// computeHash creates a hash using the FNV-1a algorithm
+func computeHash(b []byte, seed []byte) uint32 {
```

Comment:
Is there any reason to use a uint32 instead of a uint64?

Reply:
I took this code from the existing implementation in [probabilistic sampler processor](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/85198a60c315d75fcf74b6b84a6504a6e699e00a/processor/probabilisticsamplerprocessor/fnvhasher.go#L22).
My goal was to apply the same hash computation so they can be chained in the same pipeline and sample consistently.

I suggest addressing this in a followup PR which will also align the behavior for both processor by extracting the function into a common package.

Reply:
This makes sense to me.